pyiron_base/generic/filedata.py

```diff
@@ -0,0 +1,177 @@
+
+        Args:
+            filename (str): path to the file to be displayed.
+    """
+    _, filetype = os.path.splitext(filename)
+    if filetype.lower() in ['.h5', '.hdf']:
+        return FileHDFio(file_name=filename)
+    if filetype.lower() in ['.json']:
+        return _load_json(filename)
+    elif filetype.lower() in ['.txt']:
+        return _load_txt(filename)
+    elif filetype.lower() in ['.csv']:
+        return _load_csv(filename)
+    elif _has_imported['PIL'] and filetype.lower() in Image.registered_extensions():
+        return _load_img(filename)
+    else:
+        return _load_default(filename)
+
+
+def _load_txt(file):
+    with open(file) as f:
+        return f.readlines()
+
+
+def _load_json(file):
+    with open(file) as f:
+        return json.load(f)
+
+
+def _load_csv(file):
+    return pandas.read_csv(file)
+
+
+def _load_img(file):
+    return Image.open(file)
+
+
+def _load_default(file):
+    try:
+        return _load_txt(file)
+    except UnicodeDecodeError:
+        return file
+
+
+class FileData:
+
+    """ FileData stores an instance of a data file, e.g. a single Image from a measurement """
+    def __init__(self, source=None, data=None, filename=None, metadata=None, filetype=None):
+        """
+            FileData class to store data and associated metadata.
+
+            Args:
+                source (str/None): path to the data file
+                data (object/None): object containing data
+                filename (str/None): filename associated with the data object, Not used if source is given!
+                metadata (dict/InputList): Dictionary of metadata associated with the data
+                filetype (str): File extension associated with the type data,
+                                If provided this overwrites the assumption based on the extension of the filename.
+        """
+        if (source is None) and (data is None):
+            raise ValueError("No data given")
+        self._data = data
+        if source is not None:
+            self.filename = os.path.split(source)[1]
+            self.source = source
+        elif filename is None:
+            raise ValueError("No filename given")
+        else:
+            self.filename = filename
+        if (filetype is None) and (self.filename is not None):
+            filetype = os.path.splitext(self.filename)[1]
+            if len(filetype[1:]) == 0:
+                self.filetype = None
+            else:
+                self.filetype = filetype[1:]
+        else:
+            self.filetype = filetype
+        if metadata is None:
+            self.metadata = {}
+        else:
+            self.metadata = metadata
+        self._hasdata = True if self._data is not None else False
+
+    @property
+    def data(self):
+        """Return the associated data."""
+        if self._hasdata:
+            return self._data
+        else:
+            return load_file(self.source)
+
+
+class DisplayItem:
```

Comment:
Maybe I missed it, but where is this class used?

Reply:
Not at all any more, I'll throw it out.